sdk/python/kfp/utils.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random 
+import string
+
+def use_secret(secret_name:str, secret_volume_mount_path:str, env_variable:str=None, secret_file_path_in_volume:str=None):
+    """    
+       An operator that configures the container to use a secret.
+       
+       This assumes that the secret is created and availabel in the k8s cluster.
+    
+    Keyword Arguments:
+        secret_name {String} -- [Required] The k8s secret name.
+        secret_volume_mount_path {String} -- [Required] The path to the secret that is mounted.
+        env_variable {String} -- Env variable pointing to the mounted secret file. Requires both the env_variable and secret_file_path_in_volume to be defined. 
+                                 The value is the path to the secret.
+        secret_file_path_in_volume {String} -- The path to the secret in the volume. This will be the value of env_variable. 
+                                 Both env_variable and secret_file_path_in_volume needs to be set if any env variable should be created.
+    
+    Raises:
+        ValueError: If not the necessary variables (secret_name, volume_name", secret_volume_mount_path) are supplied.
+                    Or only one of  env_variable and secret_file_path_in_volume are supplied
+    
+    Returns:
+        [ContainerOperator] -- Returns the container operator after it has been modified. 
+    """
+
+    volume_name = ''.join(random.choices(string.ascii_lowercase + string.digits, k=10)) + "_volume"
```

Comment:
What do you think about this:
```
secret_name = str(secret_name)
if '{{' in secret_name:
    volume_name = ''.join(random.choices(string.ascii_lowercase + string.digits, k=10)) + "_volume"
else:
    volume_name = secret_name
```

Reply:
Why do we check for '{{' in the secret_name i don't know all the forbidden characters in k8s but guess there are more cases than this. Asking to learn  :)

Reply:
Will update the code in the mean time :)

Reply:
> Why do we check for '{{' in the secret_name i don't know all the forbidden characters in k8s but guess there are more cases than this. Asking to learn :).

`{{` are certainly forbidden, but that's not the issue here.
This special handling was for cases when the secret name is parametrized (although we discourage that). Parametrized volume names might work, but I just wanted to be safe.

Reply:
Another two cents: if the name is forbidden the user will be warned when submitting the pipeline package in a very straightforward way. Then it should be very easy for them to fix such issue.